### PR TITLE
Check for existing FB connection before making a new one.

### PR DESCRIPTION
**Pull request by knowuh**
I expanded this PR to address a few more issues:

### Changes
* Checks if the application already has an initialized DB connection before making a new one. 7f28beb
* Accurately track pending FireStore promises 8c63116
* Send student data in batches  7dab3a7
* Increase the visual progress indicator for students b5aa9fc

### Stories: 
* https://www.pivotaltracker.com/story/show/177479032
* https://www.pivotaltracker.com/story/show/177646735

----

### CODE REVIEW

**Review by scytacki — APPROVED** [review]
Seems OK to me.

**Review by dougmartin — APPROVED** [review]
Looks good :+1:

File: src/components/activity-completion/report-backup-options.tsx [hunk 1]
```diff
@@ -49,24 +59,36 @@ export class ReportBackupOptions extends React.PureComponent<IProps, IState> {
       : () => null;
 
     const reportButtonLabel = sending
-      ? "sending ..."
-      : "Report My Work";
+      ? <>Sending...  &nbsp; <IconSpin className="progress"/></>
+      : <>Report My Work</>;
```
Comment (scytacki) [inline src/components/activity-completion/report-backup-options.tsx:63]: Could remove this.

**Review by scytacki — CHANGES_REQUESTED** [review]
remove some redundant code.

**Review by scytacki — APPROVED** [review]
Looks good.

**Pull request merged**